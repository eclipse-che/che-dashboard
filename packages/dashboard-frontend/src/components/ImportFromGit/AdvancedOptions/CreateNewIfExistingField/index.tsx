/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, Switch } from '@patternfly/react-core';
import React from 'react';

export type Props = {
  onChange: (createNewIfExisting: boolean | undefined) => void;
  createNewIfExisting: boolean | undefined;
};
export type State = {
  createNewIfExisting: boolean;
};

export class CreateNewIfExistingField extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      createNewIfExisting: this.props.createNewIfExisting || false,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    const createNewIfExisting = this.props.createNewIfExisting || false;
    if (prevProps.createNewIfExisting !== createNewIfExisting) {
      this.setState({ createNewIfExisting });
    }
  }

  private handleChange(createNewIfExisting: boolean) {
    this.setState({ createNewIfExisting });
    this.props.onChange(createNewIfExisting);
  }

  public render() {
    const { createNewIfExisting } = this.state;

    return (
      <FormGroup label="Create New If Existing">
        <Switch
          id="create-new-if-existing-switch"
          aria-label="Create New If Existing"
          isChecked={createNewIfExisting}
          onChange={value => this.handleChange(value)}
        />
      </FormGroup>
    );
  }
}
