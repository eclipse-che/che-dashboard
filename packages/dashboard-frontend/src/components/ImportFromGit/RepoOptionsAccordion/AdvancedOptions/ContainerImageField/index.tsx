/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, TextInput } from '@patternfly/react-core';
import React from 'react';

export type Props = {
  onChange: (definition: string | undefined) => void;
  containerImage: string | undefined;
};
export type State = {
  containerImage: string | undefined;
};

export class ContainerImageField extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      containerImage: this.props.containerImage,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    const { containerImage } = this.props;
    if (prevProps.containerImage !== containerImage) {
      this.setState({ containerImage });
    }
  }

  private handleChange(value: string) {
    let containerImage: string | undefined = value.trim();
    containerImage = containerImage !== '' ? containerImage : undefined;
    if (containerImage !== this.state.containerImage) {
      this.setState({ containerImage });
      this.props.onChange(containerImage);
    }
  }

  public render() {
    const containerImage = this.state.containerImage || '';

    return (
      <FormGroup label="Container Image">
        <TextInput
          aria-label="Container Image"
          placeholder="Enter the container image"
          onChange={value => this.handleChange(value)}
          value={containerImage}
        />
      </FormGroup>
    );
  }
}
