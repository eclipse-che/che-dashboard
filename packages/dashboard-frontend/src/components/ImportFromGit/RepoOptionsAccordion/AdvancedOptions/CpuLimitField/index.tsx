/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, Slider } from '@patternfly/react-core';
import React from 'react';

const steps = [
  { value: 0, label: 'default' },
  { value: 1, label: '1' },
  { value: 2, label: '2', isLabelHidden: true },
  { value: 3, label: '3', isLabelHidden: true },
  { value: 4, label: '4' },
  { value: 5, label: '5', isLabelHidden: true },
  { value: 6, label: '6', isLabelHidden: true },
  { value: 7, label: '7', isLabelHidden: true },
  { value: 8, label: '8' },
];

export type Props = {
  onChange: (cpuLimit: number) => void;
  cpuLimit: number;
};
export type State = {
  cpuLimit: number;
};

export class CpuLimitField extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      cpuLimit: this.props.cpuLimit,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    const { cpuLimit } = this.props;
    if (prevProps.cpuLimit !== cpuLimit && cpuLimit !== this.state.cpuLimit) {
      this.setState({ cpuLimit });
    }
  }

  private handleChange(cpuLimit: number) {
    if (cpuLimit !== this.state.cpuLimit) {
      this.setState({ cpuLimit });
      this.props.onChange(cpuLimit);
    }
  }

  private getLabel(cpuLimit: number): string {
    const label = 'CPU Limit';
    if (cpuLimit === 0) {
      return label;
    } else if (cpuLimit === 1) {
      return `${label} (1 core)`;
    }

    return `${label} (${cpuLimit} cores)`;
  }

  public render() {
    const cpuLimit = this.state.cpuLimit;
    const label = this.getLabel(cpuLimit);

    return (
      <FormGroup label={label}>
        <Slider
          data-testid="cpu-limit-slider"
          value={cpuLimit}
          onChange={value => this.handleChange(value)}
          max={steps[steps.length - 1].value}
          customSteps={steps}
        />
      </FormGroup>
    );
  }
}
