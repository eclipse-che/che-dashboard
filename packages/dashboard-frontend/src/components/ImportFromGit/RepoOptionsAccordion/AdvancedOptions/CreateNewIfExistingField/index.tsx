/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, Switch } from '@patternfly/react-core';
import React from 'react';

import { Navigation } from '@/Layout/Navigation';
import { CREATE_NEW_IF_EXIST_SWITCH_ID } from '@/pages/GetStarted/SamplesList/Toolbar/CreateNewIfExistSwitch';

export type Props = {
  onChange: (createNewIfExisting: boolean | undefined) => void;
  createNewIfExisting: boolean | undefined;
  isHidden?: boolean;
};
export type State = {
  createNewIfExisting: boolean;
};

export class CreateNewIfExistingField extends React.PureComponent<Props, State> {
  private readonly subscribeCallback: (isChecked: boolean | undefined) => void;
  constructor(props: Props) {
    super(props);

    this.state = {
      createNewIfExisting: this.props.createNewIfExisting || false,
    };

    this.subscribeCallback = (isChecked: boolean | undefined) =>
      this.handleChange(isChecked === true);
  }

  public componentDidMount() {
    Navigation.pageState[CREATE_NEW_IF_EXIST_SWITCH_ID] = {
      isChecked: this.state.createNewIfExisting,
    };
    // Subscribe to the page state when the component is mounted
    Navigation.pageState.subscribe(this.subscribeCallback, CREATE_NEW_IF_EXIST_SWITCH_ID);
  }

  public componentWillUnmount() {
    // Unsubscribe from the page state when the component is unmounted
    Navigation.pageState.unsubscribe(this.subscribeCallback, CREATE_NEW_IF_EXIST_SWITCH_ID);
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    const createNewIfExisting = this.props.createNewIfExisting || false;
    if (prevProps.createNewIfExisting !== createNewIfExisting) {
      this.setState({ createNewIfExisting });
    }
  }

  private handleChange(createNewIfExisting: boolean): void {
    if (createNewIfExisting === this.state.createNewIfExisting) {
      return; // No change, do nothing
    }
    this.setState({ createNewIfExisting });
    this.props.onChange(createNewIfExisting);
  }

  public render() {
    const { createNewIfExisting } = this.state;
    const display = this.props.isHidden ? 'none' : 'block';

    return (
      <FormGroup label="Create New If Existing" style={{ display }}>
        <Switch
          id="advanced-options-create-new-if-exist-switch"
          aria-label="Create New If Existing"
          isChecked={createNewIfExisting}
          onChange={value => this.handleChange(value)}
        />
      </FormGroup>
    );
  }
}
