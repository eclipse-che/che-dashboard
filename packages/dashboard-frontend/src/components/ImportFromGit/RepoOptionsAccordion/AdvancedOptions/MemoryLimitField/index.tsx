/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, Slider } from '@patternfly/react-core';
import React from 'react';

import { formatBytes } from '@/components/ImportFromGit/helpers';

export const STEP = 1073741824;

const steps = [
  { value: 0, label: 'default' },
  { value: 1, label: '1', isLabelHidden: true },
  { value: 2, label: '2', isLabelHidden: true },
  { value: 3, label: '4', isLabelHidden: true },
  { value: 4, label: '4' },
  { value: 6, label: '6', isLabelHidden: true },
  { value: 8, label: '8', isLabelHidden: true },
  { value: 12, label: '12', isLabelHidden: true },
  { value: 16, label: '16' },
  { value: 20, label: '20', isLabelHidden: true },
  { value: 24, label: '24', isLabelHidden: true },
  { value: 28, label: '28', isLabelHidden: true },
  { value: 32, label: '32' },
];

export type Props = {
  onChange: (memoryLimit: number) => void;
  memoryLimit: number;
};
export type State = {
  memoryLimit: number;
};

export class MemoryLimitField extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    const memoryLimit = this.getMemoryLimit();

    this.state = {
      memoryLimit,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    if (prevProps.memoryLimit !== this.props.memoryLimit) {
      const memoryLimit = this.getMemoryLimit();
      if (memoryLimit !== this.state.memoryLimit) {
        this.setState({ memoryLimit });
      }
    }
  }

  private handleChange(memoryLimit: number) {
    if (memoryLimit !== this.state.memoryLimit) {
      this.setState({ memoryLimit });
      this.props.onChange(memoryLimit * STEP);
    }
  }

  private getMemoryLimit(): number {
    const memoryLimit = this.props.memoryLimit;
    if (memoryLimit <= STEP) {
      return 0;
    }
    return memoryLimit / STEP;
  }

  private getLabel(memoryLimit: number): string {
    if (memoryLimit > 0) {
      return `Memory Limit (${formatBytes(memoryLimit * STEP)})`;
    }

    return 'Memory Limit';
  }

  public render() {
    const memoryLimit = this.state.memoryLimit;
    const label = this.getLabel(memoryLimit);

    return (
      <FormGroup label={label}>
        <Slider
          data-testid="memory-limit-slider"
          value={memoryLimit}
          onChange={value => this.handleChange(value)}
          max={steps[steps.length - 1].value}
          customSteps={steps}
        />
      </FormGroup>
    );
  }
}
