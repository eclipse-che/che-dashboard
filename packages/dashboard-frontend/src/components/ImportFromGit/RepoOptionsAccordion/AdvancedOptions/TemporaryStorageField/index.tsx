/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, Switch } from '@patternfly/react-core';
import React from 'react';

export type Props = {
  onChange: (isTemporary: boolean | undefined) => void;
  isTemporary: boolean | undefined;
};
export type State = {
  isTemporary: boolean;
};

export class TemporaryStorageField extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      isTemporary: this.props.isTemporary || false,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    const isTemporary = this.props.isTemporary || false;
    if (prevProps.isTemporary !== isTemporary) {
      this.setState({ isTemporary });
    }
  }

  private handleChange(isTemporary: boolean) {
    this.setState({ isTemporary });
    this.props.onChange(isTemporary);
  }

  public render() {
    const { isTemporary } = this.state;

    return (
      <FormGroup label="Temporary Storage">
        <Switch
          id="temporary-storage-switch"
          aria-label="Temporary Storage"
          isChecked={isTemporary}
          onChange={value => this.handleChange(value)}
        />
      </FormGroup>
    );
  }
}
