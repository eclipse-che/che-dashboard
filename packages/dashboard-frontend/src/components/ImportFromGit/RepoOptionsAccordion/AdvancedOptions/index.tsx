/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Form } from '@patternfly/react-core';
import React from 'react';

import { ContainerImageField } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/ContainerImageField';
import { CpuLimitField } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/CpuLimitField';
import { CreateNewIfExistingField } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/CreateNewIfExistingField';
import { MemoryLimitField } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/MemoryLimitField';
import { TemporaryStorageField } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/TemporaryStorageField';

export type Props = {
  containerImage: string | undefined;
  temporaryStorage: boolean | undefined;
  createNewIfExisting: boolean | undefined;
  memoryLimit: number | undefined;
  cpuLimit: number | undefined;
  onChange: (
    containerImage: string | undefined,
    temporaryStorage: boolean | undefined,
    createNewIfExisting: boolean | undefined,
    memoryLimit: number | undefined,
    cpuLimit: number | undefined,
  ) => void;
};

export type State = {
  containerImage: string | undefined;
  temporaryStorage: boolean | undefined;
  createNewIfExisting: boolean | undefined;
  memoryLimit: number | undefined;
  cpuLimit: number | undefined;
};

export class AdvancedOptions extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      containerImage: props.containerImage,
      temporaryStorage: props.temporaryStorage,
      createNewIfExisting: props.createNewIfExisting,
      memoryLimit: props.memoryLimit,
      cpuLimit: props.cpuLimit,
    };
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    const { containerImage, temporaryStorage, createNewIfExisting, memoryLimit, cpuLimit } =
      this.props;

    if (containerImage !== prevProps.containerImage) {
      this.setState({ containerImage });
    }

    if (temporaryStorage !== prevProps.temporaryStorage) {
      this.setState({ temporaryStorage });
    }

    if (createNewIfExisting !== prevProps.createNewIfExisting) {
      this.setState({ createNewIfExisting });
    }

    if (memoryLimit !== prevProps.memoryLimit) {
      this.setState({ memoryLimit });
    }

    if (cpuLimit !== prevProps.cpuLimit) {
      this.setState({ cpuLimit });
    }
  }

  private handleContainerImage(containerImage: string | undefined) {
    const { temporaryStorage, createNewIfExisting, memoryLimit, cpuLimit } = this.state;

    this.setState({ containerImage });
    this.props.onChange(
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
    );
  }

  private handleTemporaryStorage(temporaryStorage: boolean | undefined) {
    const { containerImage, createNewIfExisting, memoryLimit, cpuLimit } = this.state;

    this.setState({ temporaryStorage });
    this.props.onChange(
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
    );
  }

  private handleCreateNewIfExisting(createNewIfExisting: boolean | undefined) {
    const { containerImage, temporaryStorage, memoryLimit, cpuLimit } = this.state;

    this.setState({ createNewIfExisting });
    this.props.onChange(
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
    );
  }

  private handleMemoryLimit(memoryLimit: number | undefined) {
    const { containerImage, temporaryStorage, createNewIfExisting, cpuLimit } = this.state;

    this.setState({ memoryLimit });
    this.props.onChange(
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
    );
  }

  private handleCpuLimit(cpuLimit: number | undefined) {
    const { containerImage, temporaryStorage, createNewIfExisting, memoryLimit } = this.state;

    this.setState({ cpuLimit });
    this.props.onChange(
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
    );
  }

  public render() {
    const { containerImage, temporaryStorage, createNewIfExisting, memoryLimit, cpuLimit } =
      this.state;
    return (
      <Form isHorizontal={true} onSubmit={e => e.preventDefault()}>
        <ContainerImageField
          onChange={containerImage => this.handleContainerImage(containerImage)}
          containerImage={containerImage}
        />
        <TemporaryStorageField
          onChange={temporaryStorage => this.handleTemporaryStorage(temporaryStorage)}
          isTemporary={temporaryStorage}
        />
        <CreateNewIfExistingField
          onChange={createNewIfExisting => this.handleCreateNewIfExisting(createNewIfExisting)}
          createNewIfExisting={createNewIfExisting}
        />
        <MemoryLimitField
          onChange={memoryLimit => this.handleMemoryLimit(memoryLimit)}
          memoryLimit={memoryLimit || 0}
        />
        <CpuLimitField
          onChange={cpuLimit => this.handleCpuLimit(cpuLimit)}
          cpuLimit={cpuLimit || 0}
        />
      </Form>
    );
  }
}
