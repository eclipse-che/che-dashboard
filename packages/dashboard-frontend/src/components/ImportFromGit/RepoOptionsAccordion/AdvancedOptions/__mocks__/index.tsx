/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions';

export class AdvancedOptions extends React.PureComponent<Props> {
  public render() {
    const {
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
      onChange,
    } = this.props;

    return (
      <div>
        <div>Advanced Options</div>
        <div data-testid="advanced-options">{`${containerImage}, ${temporaryStorage}, ${createNewIfExisting}, ${memoryLimit}, ${cpuLimit}`}</div>
        <button
          onClick={() =>
            onChange('newContainerImage', !temporaryStorage, !createNewIfExisting, 1073741824, 1)
          }
        >
          Advanced Options Change
        </button>
      </div>
    );
  }
}
