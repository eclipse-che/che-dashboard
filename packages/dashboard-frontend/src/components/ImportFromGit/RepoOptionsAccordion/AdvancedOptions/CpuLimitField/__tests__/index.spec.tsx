/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { NumberInputProps } from '@patternfly/react-core';
import { fireEvent, screen } from '@testing-library/react';
import React from 'react';

import { CpuLimitField } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/CpuLimitField';
import getComponentRenderer from '@/services/__mocks__/getComponentRenderer';

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);
// mute the outputs
console.error = jest.fn();

jest.mock('@patternfly/react-core', () => {
  return {
    ...jest.requireActual('@patternfly/react-core'),
    NumberInput: (obj: NumberInputProps) => (
      <input
        type="range"
        data-testid={obj['data-testid']}
        value={obj.value}
        onChange={event => {
          if (obj.onChange) {
            obj.onChange(event);
          }
        }}
      />
    ),
  };
});

const mockOnChange = jest.fn();

describe('CpuLimitField', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('2Gi snapshot', () => {
    const snapshot = createSnapshot(2);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should be init with 2Gi and switched to 8Gi', () => {
    renderComponent(2);
    const element = screen.getByTestId('cpu-limit-input') as HTMLInputElement;
    const getVal = () => parseInt(element.value);

    expect(element).toBeDefined();
    expect(getVal()).toEqual(2);

    fireEvent.change(element, { target: { value: 8 } });

    expect(getVal()).toEqual(8);
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});

function getComponent(cpuLimit: number) {
  return <CpuLimitField cpuLimit={cpuLimit} onChange={mockOnChange} />;
}
