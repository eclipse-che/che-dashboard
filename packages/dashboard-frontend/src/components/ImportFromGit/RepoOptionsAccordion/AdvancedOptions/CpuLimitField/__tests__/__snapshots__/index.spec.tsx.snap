// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpuLimitField 2Gi snapshot 1`] = `
<div
  className="pf-c-form__group"
>
  <div
    className="pf-c-form__group-label"
  >
    <label
      className="pf-c-form__label"
    >
      <span
        className="pf-c-form__label-text"
      >
        CPU Limit (2 cores)
      </span>
    </label>
     
  </div>
  <div
    className="pf-c-form__group-control"
  >
    <input
      data-testid="cpu-limit-slider"
      onChange={[Function]}
      type="range"
      value={2}
    />
  </div>
</div>
`;
